seacrowd/sea_datasets/thai_gpteacher/thai_gpteacher.py

```diff
@@ -0,0 +1,118 @@
+            "role_play": "https://huggingface.co/datasets/Thaweewat/gpteacher-20k-th/resolve/main/gpteacher-gpt4-instruct-roleplay-2k-th.parquet",
+        }
+    },
+}
+
+_SUPPORTED_TASKS = [Tasks.INSTRUCTION_TUNING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ThaiGPTeacherDataset(datasets.GeneratorBasedBuilder):
+    """Thai-instructed dataset translated using Google Cloud Translation from GPTeacher."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_t2t",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_t2t",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "instruction": datasets.Value("string"),
+                    "input": datasets.Value("string"),
+                    "output": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> list[datasets.SplitGenerator]:
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_dir, "split": "train"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> tuple[int, dict]:
+        df1 = pd.read_parquet(filepath[split]["qa"])
+        df2 = pd.read_parquet(filepath[split]["role_play"])
+        df = pd.concat([df1, df2], ignore_index=True)
+        if self.config.schema == "source":
+            for i, row in df.iterrows():
+                yield i, {"instruction": row["instruction"], "input": row["input"], "output": row["output"]}
+
+        elif self.config.schema == "seacrowd_t2t":
+            for i, row in df.iterrows():
+                yield i, {
+                    "id": str(i),
+                    "text_1": row["instruction"] + row["input"],
```

Comment:
I think we need to put explicit delimiter between instruction and input
because in some example, the input is not started by space, for example in id 4:
```
{'id': '4', 'text_1': ' แปลข้อความที่ตัดตอนมาจากภาษาอังกฤษเป็นภาษาฝรั่งเศสThere was a boy named Romeo and a girl named Juliet. They were from two different families who were enemies. Despite this, they fell in love with each other at their very first meeting.', 'text_2': "Il y avait un garçon nommé Roméo et une fille nommée Juliette. Ils étaient issus de deux familles différentes qui étaient ennemies. Malgré cela, ils sont tombés amoureux l'un de l'autre dès leur première rencontre.", 'text_1_name': 'instruction+input', 'text_2_name': 'output'}
```

still unsure if the space is enough or is it better to use other delimiter like `\n`